I made a prototype of the melee termite that I think feels pretty goo…

desultisoft opened the pull request:
…d to fight.

It uses several timers and pauses animation since we don't really have a tired or stop animation.
It uses kinematic based motion to do the dash.
I changed the ground effect to be able to use an alternative knockback because it made more sense for the termites to knock the player back.
I changed the sprite flip code as I think it needed to work before the attack in a way that didn't just rely on move direction.

desultisoft committed:
I made a prototype of the melee termite that I think feels pretty good to fight.

It uses several timers and pauses animation since we don't really have a tired or stop animation.
It uses kinematic based motion to do the dash.
I changed the ground effect to be able to use an alternative knockback because it made more sense for the termites to knock the player back.
I changed the sprite flip code as I think it needed to work before the attack in a way that didn't just rely on move direction.

desultisoft committed:
I changed the melee termite to be more fair and in line with Alex's requests

They now have a charge sprite frame they use for the wind up.
They now are more fair and don't lunge several body's length.
Removed health bar visibility from all termites.
All termite enemies now use layers to make sure they don't overlap.

Flamebeard10339 reviewed (changes requested):
The termite collision layers are wrong. They should look like this: 
![image](https://user-images.githubusercontent.com/7536002/141367851-3cacfb7a-2d0d-498a-9409-a832c15fcc53.png)

There is still the issue that the on death effects play slightly after they should because the termite is waiting for the dieing animation to complete. I feel that the solution is to spawn a corpse object and eschew the yielding on MeleeTermite:die():62. 

The current situation also means that the player can't walk over the corpses for the brief duration that they exist.

Also the knockback for the player seems instant. Might be something to look into later.

Also please don't commit changes to World.tscn, or project.godot unless strictly necessary. On Github desktop you can't skip these changed files by unchecking the boxes before hitting commit:
![image](https://user-images.githubusercontent.com/7536002/141368557-041a3828-fb2e-4837-8b3c-fd7522e5266f.png)

desultisoft committed:
I cleaned up some of the termite enemies and their death sequence.

Termites now spawn a corpse tscn when dying.
The corpse uses a tween to slowly fade out then queue free.
Enemy death functionality is now called right away due to the corpse taking their place.
Enemies now use a mask to collide rather than being on several layers.

desultisoft committed:
Merge branch 'main' into melee-termite

desultisoft committed:
I updated the termites a bit.

Volatile Termites now use the updated volatile art.
I removed the merge conflict with world.tscn caused by the pull rq previously.

desultisoft committed:
I re-added enemy knockback and white flash into the termite enemies.

desultisoft committed:
I Re-Added charge animation that was removed when resolving conflicts for termites.

Flamebeard10339 commented on src/Code/Enemies/Enemy.gd:
```diff
@@ -78,12 +82,13 @@ func die(ability_source):
 
 
-func _move():
+func _move(_delta):
```
_ before a variable means that the variable is not used in the function.

Flamebeard10339 commented on src/Code/Enemies/Termite/MeleeTermite.gd:
```diff
@@ -55,14 +56,28 @@ func die(abilitysource):
+	corpse.global_position = global_position
+	display.global_position = global_position
+
+	root.add_child(corpse)
```
line 69 to 79 should be in corpse.gd

Flamebeard10339 reviewed (approved):
Looks good.
